pkg/openfeature/provider.go

```diff
@@ -60,6 +65,62 @@ type StateHandler interface {
 	Status() State
 }
 
+// NoopStateHandler is a noop StateHandler implementation
+// Status always set to ReadyState to comply with specification
+type NoopStateHandler struct {
+}
+
+func (s *NoopStateHandler) Init(e EvaluationContext) {
+	// NOOP
+}
+
+func (s *NoopStateHandler) Shutdown() {
+	// NOOP
+}
+
+func (s *NoopStateHandler) Status() State {
+	return ReadyState
+}
+
+// Eventing
+
+// EventHandler is the eventing contract enforced for FeatureProvider
+type EventHandler interface {
+	EventChannel() <-chan Event
+}
+
+// EventType emitted by a provider implementation
+type EventType string
+
+// ProviderEventDetails is the event payload emitted by FeatureProvider
+type ProviderEventDetails struct {
+	Message       string
+	FlagChanges   []string
+	EventMetadata map[string]interface{}
+}
+
+// Event is an event emitted by a FeatureProvider.
+type Event struct {
+	ProviderName string
+	EventType
+	ProviderEventDetails
+}
```

Comment:
Does this type need to be exported? Unlike `ProviderEventDetails` and `EventDetails`, it's not in the spec.

Reply:
For go-sdk implementation, events are emitted from providers. And it's the OF implementation that handles this event. This event emitting is done through a channel that eventing providers expose [1] and this is the value type channel emits. This is why this needs to be exposed. 

[1] - https://github.com/open-feature/go-sdk/blob/8ee7d3ce4b1dc81669b2bdf0fe74e6ac6cf795c3/pkg/openfeature/provider.go#L89